python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -204,7 +205,7 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info):
     vocabulary_df = file_io.read_df(feature_layer_info["args"]["vocabulary_file"])
     vocabulary_keys = vocabulary_df["key"].values
     vocabulary_ids = (
-        vocabulary_df["id"].values if "id" in vocabulary_df else range(len(vocabulary_keys))
+        vocabulary_df["id"].values if "id" in vocabulary_df else list(range(len(vocabulary_keys)))
```

Comment:
@darshshah This was a syntax error. Had to be a list, but was range instead. The reason it wasn't caught is because it works for training and testing and only throws error when saving.